core/src/main/java/org/openconnectors/connect/SinkConnector.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.openconnectors.connect;
+
+import org.openconnectors.config.Connector;
+import org.openconnectors.util.SinkConnectorContext;
+
+import java.util.Collection;
+
+public abstract class SinkConnector<T extends SinkConnectorContext, U> extends Connector<T> {
+
+    public abstract void publish(final Collection<U> messages) throws Exception;
+
+    public abstract void flush() throws Exception;
+
```

Comment:
Could flush throw non available type exception?

Reply:
I am keeping the exception type general as of now , will specialize the type as the design matures.